lib/MemoryModel/PointerAnalysisImpl.cpp

```diff
@@ -219,7 +220,12 @@ void BVDataPTAImpl::dumpAllPts() {
 void BVDataPTAImpl::onTheFlyCallGraphSolve(const CallSiteToFunPtrMap& callsites, CallEdgeMap& newEdges) {
     for(CallSiteToFunPtrMap::const_iterator iter = callsites.begin(), eiter = callsites.end(); iter!=eiter; ++iter) {
         const CallBlockNode* cs = iter->first;
-        if (isVirtualCallSite(cs->getCallSite())) {
+
+        // If we have a DCHG (i.e., ctir available), use that to check for virtualness.
+        bool isVirtual = LLVMModuleSet::getLLVMModuleSet()->allCTir() ?
```

Comment:
Can we move this into method 'isVirtualCallSite'?

Reply:
isVirtualCallsite would need access to the chg, which would require another argument. Is this desired?

Reply:
> Maybe we can put `isVirtualCallSite` in the `CommonCHGraph` and move CPPUtil's `isVirtualCallSite` to `CHA`

There is a call to `cppUtil::isVirtualCallSite` in `DDAClient.cpp`. It seems to me there is no `CHGraph`. Is that right or am I missing something?

Reply:
There are few places invoking `isVirtualCallSite`, that is why we need to make a single wrapper for it, otherwise, it may cause crashes in other places.

Or we can move DCHG's `isVirtualCallSite` into `cppUtil::isVirtualCallSite`?

Reply:
> Or we can move DCHG's `isVirtualCallSite` into `cppUtil::isVirtualCallSite`?

This seems to be the best way.